src/devtools/packages/devtools-reps/object-inspector/components/ObjectInspector.tsx

```diff
@@ -72,6 +72,20 @@ class OI extends PureComponent<ObjectInspectorProps> {
     this.activeItem = this.props.activeItem;
   }
 
+  componentDidMount() {
+    this.ensureRootsLoaded();
+  }
+
+  async ensureRootsLoaded() {
+    const rootsToLoad = this.getRoots().filter(
+      (root): root is ValueItem => root.type === "value" && !root.loaded
+    );
+    if (rootsToLoad.length > 0) {
+      await Promise.all(rootsToLoad.map(root => root.contents.load()));
+      this.forceUpdate();
+    }
+  }
```

Comment:
> - when the ObjectInspector is rendered with a root `ValueItem` that hasn't been loaded yet, it will load it now and then rerender (ideally this should never happen: the ObjectInspector should only be rendered with a root `ValueItem` that has already been loaded, but that isn't always the case right now and I decided to add this logic to make the ObjectInspector more resilient)

I guess this relates to what you were saying in the PR description :grin:

Should we add some kind of warning here? I think if this happens, maybe it suggests that the higher level code that's orchestrating the loading of value items isn't working correct? and this "fixes" things but also in a way that is less efficient because of the loading waterfall (load a little data and render, then load more data and re-render).

Reply:
I haven't added a warning so far because I already know that the console code doesn't handle this correctly. I'll take another look if I can fix this (and if so, add a warning here).

Reply:
Just my two cents: it seems like adding a warning and TODO's at the places we already know that violate this would be enough to get this in without having to fix any additional issues. Even though this change is small it's already pretty dense!